lib/assets/javascripts/manage-prototype/manage-plugins.test.js

```diff
@@ -0,0 +1,225 @@
+/**
+ * @jest-environment jsdom
+ */
+
+async function fetchResponse (data) {
+  return {
+    json: async () => data
+  }
+}
+
+const loadedHTML = `
+      <div>
+        <div id="panel-manual-instructions"></div>
+        <div id="panel-processing" hidden=""></div>
+        <div id="panel-complete" hidden=""></div>
+        <div id="panel-error" hidden=""></div>
+        <div id="instructions-complete" hidden=""></div>
+      </div>
+    `
+
+const processingHTML = `
+      <div>
+        <div id="panel-manual-instructions" hidden=""></div>
+        <div id="panel-processing"></div>
+        <div id="panel-complete" hidden=""></div>
+        <div id="panel-error" hidden=""></div>
+        <div id="instructions-complete" hidden=""></div>
+      </div>
+    `
+
+const completedHTML = `
+      <div>
+        <div id="panel-manual-instructions" hidden=""></div>
+        <div id="panel-processing" hidden=""></div>
+        <div id="panel-complete"></div>
+        <div id="panel-error" hidden=""></div>
+        <div id="instructions-complete"></div>
+      </div>
+    `
+
+const errorHTML = `
+      <div>
+        <div id="panel-manual-instructions" hidden=""></div>
+        <div id="panel-processing" hidden=""></div>
+        <div id="panel-complete" hidden=""></div>
+        <div id="panel-error"></div>
+        <div id="instructions-complete" hidden=""></div>
+      </div>
+    `
+
+describe('manage-plugins', () => {
+  global.fetch = jest.fn().mockResolvedValue(null)
+
+  const performActionUrl = '/manage-prototype/plugins/'
+  const pollStatusUrl = '/manage-prototype/plugins//status'
+
+  const { document, window } = global
+  let managePlugins
+  let fetchList
+
+  const mockFetch = (statuses) => {
+    let responseIndex = 0
+    const responses = statuses.map((status) => {
+      return status === 'throws'
```

Comment:
I think this would be better written as a function with an `if`:

```
statuses.map((status) => () => {
  fetchList.push(url)
  if (status === 'throws') {
    return Promise.reject(new Error())
  }
  return fetchResponse({ status })
})
```

Reply:
Done as requested